tests/v2/validation/clustertemplates/provisioning_clustertemplates.go

```diff
@@ -0,0 +1,96 @@
+package clustertemplates
+
+import (
+	"github.com/rancher/rancher/tests/v2/actions/clusters"
+	"github.com/rancher/rancher/tests/v2/actions/clustertemplates"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/shepherd/clients/rancher"
+	mgmt "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+)
+
+const (
+	CniCalico                        = "calico"
+	ClusterEnforcementSetting        = "cluster-template-enforcement"
+	EnabledClusterEnforcementSetting = "true"
+	IsRequiredQuestion               = true
+	UserPrincipalID                  = "local://"
+)
+
+var (
+	Questions = []mgmt.Question{
+		0: {
+			Variable: "rancherKubernetesEngineConfig.kubernetesVersion",
+			Default:  "",
+			Required: IsRequiredQuestion,
+			Type:     "string",
+		},
+		1: {
+			Variable: "rancherKubernetesEngineConfig.network.plugin",
+			Default:  "",
+			Required: IsRequiredQuestion,
+			Type:     "string",
+		},
+		2: {
+			Variable: "rancherKubernetesEngineConfig.services.etcd.backupConfig.intervalHours",
+			Default:  "",
+			Required: IsRequiredQuestion,
+			Type:     "string",
+		},
+	}
+)
+
+func CreateRkeTemplate(client *rancher.Client, members []mgmt.Member) (*mgmt.ClusterTemplate, error) {
+	rkeTemplateName := mgmt.ClusterTemplate{
+		Name:    namegenerator.AppendRandomString("rketemplate-"),
+		Members: members,
+	}
+
+	createTemplate, err := client.Management.ClusterTemplate.Create(&rkeTemplateName)
+	return createTemplate, err
+}
+
+func CreateRkeTemplateRevision(client *rancher.Client, templateRevisionConfig mgmt.ClusterTemplateRevision, templateId string) (*mgmt.ClusterTemplateRevision, error) {
+	rkeTemplateConfig := clustertemplates.NewRKE1ClusterTemplateRevisionTemplate(templateRevisionConfig, templateId)
+
+	clusterTemplateRevision, err := client.Management.ClusterTemplateRevision.Create(&rkeTemplateConfig)
+	if err != nil {
+		return nil, err
+	}
+	return clusterTemplateRevision, nil
+}
+
+func UpdateClusterTemplateClusterConfig(templateID, revisionID string, answers *mgmt.Answer) *mgmt.Cluster {
```

Comment:
this sounds like an update to the clusterTemplate would be returned, but its a constructor to return a management cluster.
but aside, I think it would be best to get rid of this func altogether and create the object on the fly since no additional logic is required.

Reply:
I understand, but the primary reason I included it is to improve the test readability and maintain a clean flow. By having an encapsulation to the object creation within a function, we can avoid passing multiple parameters throughout the test, which causes confusion.

Even though there’s no additional logic here, this function serves as an entry point for creating the cluster, making the test more maintainable and easier to follow. If we need to update or extend this logic later, we can do so in one place without impacting the test flow elsewhere.

Reply:
having a constructor like this is more confusing, just in this example I am confused on what it should be doing based on the name 😓

Reply:
I understand that the constructor might seem confusing at first glance, but it's actually used in every test throughout the code. Instead of defining the same logic in each test, it's more efficient to encapsulate this part into a dedicated function. This not only reduces redundancy but also keeps the code cleaner and more maintainable. The overall flow of the code becomes is easier to update if changes are needed in the future.

Reply:
I disagree, maybe we can take it up in a design meeting with the team tomorrow

Reply:
Yes, let's bring this up in the design meeting tomorrow. It seems like we’re coming at this from different angles, so discussing it with the team will help us align.